filesource/filesource.go

```diff
@@ -402,42 +430,16 @@ func (r *reader) makeParseConfig(obj ObjectInfo) *parser.Config {
 }
 
 func (r *reader) emit(lines []json.RawMessage) error {
-	// Message which wraps Records we'll generate.
-	var wrapper = &airbyte.Message{
-		Type: airbyte.MessageTypeRecord,
-		Record: &airbyte.Record{
-			Stream:    r.prefix,
-			EmittedAt: time.Now().UTC().UnixNano() / int64(time.Millisecond),
-		},
-	}
-	// Message which wraps State checkpoints we'll generate.
-	// Use a custom struct that matches the shape of airbyte.Message,
-	// but specializes State.Data to our type and avoids a double-encoding.
-	var stateWrapper = &struct {
-		Type  airbyte.MessageType `json:"type"`
-		State struct {
-			Data States `json:"data"`
-		} `json:"state,omitempty"`
-	}{
-		Type: airbyte.MessageTypeState,
-	}
-
-	r.shared.mu.Lock()
-	defer r.shared.mu.Unlock()
-
 	// Write out all records, wrapped in |wrapper|.
 	for _, line := range lines {
-		wrapper.Record.Data = line
-		if err := r.shared.enc.Encode(wrapper); err != nil {
+		if err := r.stream.Documents(r.binding, line); err != nil {
 			return err
 		}
 	}
 
-	// Update and write out state.
-	r.shared.states[r.prefix] = r.state
-	stateWrapper.State.Data = r.shared.states
-
-	if err := r.shared.enc.Encode(stateWrapper); err != nil {
+  if encodedCheckpoint, err := json.Marshal(r.shared.states); err != nil {
```

Comment:
Previously we would aquire the lock for the shared state and update the shared state with the state for this reader before emitting the state. Unless I'm missing something we still need to do that, since I don't see that the shared state is being updated anywhere else.